homeassistant/components/nws/weather.py

```diff
@@ -302,38 +261,35 @@ def _forecast(
     @callback
     def _async_forecast_hourly(self) -> list[Forecast] | None:
         """Return the hourly forecast in native units."""
-        return self._forecast(self._forecast_hourly, HOURLY)
+        return self._forecast(self.nws.forecast_hourly, HOURLY)
 
     @callback
     def _async_forecast_twice_daily(self) -> list[Forecast] | None:
         """Return the twice daily forecast in native units."""
-        return self._forecast(self._forecast_twice_daily, DAYNIGHT)
+        return self._forecast(self.nws.forecast, DAYNIGHT)
 
     @property
     def available(self) -> bool:
         """Return if state is available."""
-        last_success = (
-            self.coordinator.last_update_success
-            and self.coordinator_forecast_legacy.last_update_success
-        )
         if (
-            self.coordinator.last_update_success_time
-            and self.coordinator_forecast_legacy.last_update_success_time
+            self.coordinator.last_update_success
+            and self.coordinator.last_update_success_time
         ):
             last_success_time = (
                 utcnow() - self.coordinator.last_update_success_time
                 < OBSERVATION_VALID_TIME
-                and utcnow() - self.coordinator_forecast_legacy.last_update_success_time
-                < FORECAST_VALID_TIME
             )
         else:
             last_success_time = False
-        return last_success or last_success_time
+        return self.coordinator.last_update_success or last_success_time
 
     async def async_update(self) -> None:
         """Update the entity.
 
         Only used by the generic entity update service.
         """
         await self.coordinator.async_request_refresh()
-        await self.coordinator_forecast_legacy.async_request_refresh()
+        if self.forecast_coordinators["twice_daily"] is not None:
```

Comment:
This sort of behavior would be better in `CoordinatorWeatherEntity` in my opinion.